packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -126,14 +139,37 @@ export class SelectedNetworkController extends BaseController<
       state,
     });
     this.#registerMessageHandlers();
+
+    // if the network for metamask is not set, set it to the currently selected network upon initialization
+    if (this.state.domains[METAMASK_DOMAIN] === undefined) {
+      const { selectedNetworkClientId } = this.messagingSystem.call(
+        'NetworkController:getState',
+      );
+      this.setNetworkClientIdForDomain(
+        METAMASK_DOMAIN,
+        selectedNetworkClientId,
+      );
+    }
```

Comment:
```suggestion
      this.setNetworkClientIdForMetamask(selectedNetworkClientId);
    }
```

Reply:
No longer relevant as of https://github.com/MetaMask/core/pull/3908/commits/c78443841150f1f0acf3007fa9d67fae9bce5481